File: Logic Building/2_LOOPING & PATTERNS/4_Pattern Printing/24/24.js
// Q24: Print hollow diamond star pattern.
// Input: An integer n
// Output: Hollow diamond with stars

const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin });
rl.on('line', (line) => {
    const n = parseInt(line);
    for (let i = 1; i <= n; i++) {
        let row = " ".repeat(n - i);
        for (let j = 0; j < 2 * i - 1; j++) {
            row += (j === 0 || j === 2 * i - 2) ? "*" : " ";
        }
        console.log(row);
    }
    for (let i = n - 1; i >= 1; i--) {
        let row = " ".repeat(n - i);
        for (let j = 0; j < 2 * i - 1; j++) {
            row += (j === 0 || j === 2 * i - 2) ? "*" : " ";
        }
        console.log(row);
    }
    rl.close();
});

File: Logic Building/2_LOOPING & PATTERNS/4_Pattern Printing/24/24.py
# Q24: Print hollow diamond star pattern.
# Input: An integer n
# Output: Hollow diamond with stars

n = int(input())
for i in range(1, n + 1):
    row = " " * (n - i)
    for j in range(2 * i - 1):
        if j == 0 or j == 2 * i - 2:
            row += "*"
        else:
            row += " "
    print(row)
for i in range(n - 1, 0, -1):
    row = " " * (n - i)
    for j in range(2 * i - 1):
        if j == 0 or j == 2 * i - 2:
            row += "*"
        else:
            row += " "
    print(row)
